From 9b1f2561a14c9dcd680912aa341ac7518600e663 Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Sat, 29 Jun 2013 11:57:07 +0800
Subject: [PATCH] rk fb: set scaler after lcdc timing modification in one lcdc
 dual display mode        this is to mach jettb vif sync requirement

---
 drivers/video/rockchip/rk_fb.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e9e6c85c854c..b3538c1c557e 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -917,10 +917,7 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 				dev_drv->screen0->sscreen_get(dev_drv->screen0,
 					dev_drv->cur_screen->hdmi_resolution);
 			}
-			if(dev_drv->screen0->sscreen_set)
-			{
-				dev_drv->screen0->sscreen_set(dev_drv->screen0,enable);
-			}
+			
 			
 		}
 		else
@@ -930,10 +927,7 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 			dev_drv->cur_screen = dev_drv->screen0;
 			dev_drv->screen_ctr_info->set_screen_info(dev_drv->cur_screen,
 			dev_drv->screen_ctr_info->lcd_info);
-			if(dev_drv->screen0->sscreen_set)
-			{
-				dev_drv->screen0->sscreen_set(dev_drv->screen0,0);
-			}
+			
 			
 		}
 	}
@@ -1006,6 +1000,13 @@ int rk_fb_switch_screen(rk_screen *screen ,int enable ,int lcdc_id)
 		info->fbops->fb_pan_display(hdmi_var,info);
 	#endif 
 	info->fbops->fb_ioctl(info,RK_FBIOSET_CONFIG_DONE,0);
+	if(dev_drv->screen1) //for one lcdc use  scaler for dual dispaly
+	{
+		if(dev_drv->screen0->sscreen_set)
+		{
+			dev_drv->screen0->sscreen_set(dev_drv->screen0,enable);
+		}
+	}
 #if defined(CONFIG_NO_DUAL_DISP)  //close backlight for device whic do not support dual display
 	if(!enable)
 		rk29_backlight_set(1);
-- 
2.35.3

